anywidget/widget.py

```diff
@@ -22,6 +37,9 @@ def __init__(self, *args, **kwargs) -> None:
             for k in ("_esm", "_module", "_css")
             if hasattr(self, k) and not self.has_trait(k)
         }
+        # show default _esm if not defined
+        if all(i not in anywidget_traits for i in ("_esm", "_module")):
+            anywidget_traits["_esm"] = t.Unicode(DEFAULT_ESM).tag(sync=True)
```

Comment:
I think this would override `_esm` if the trait is explicitly registered via the `HasTraitsMetaclass`, e.g.,

```python
class CustomWidget(anywidget.AnyWidget):
    _esm = traitlets.Unicode(" ... ").tag(sync=True) # `anywidget_traits` will not include `_esm` or `_module` since `self.hastrait("_esm")` is true
```

Maybe we could move this logic after `self.add_traits(**anywidget_traits)`:

```python
if all(not self.has_trait(i) for i in ("_esm", "_module")):
    self.add_trait(_esm=t.Unicode(DEFAULT_ESM).tag(sync=True))
```

Reply:
I think the `test_basic` unit test should be failing...

Reply:
Ahh actually the unit tests don't assert the ESM is correct. Would you mind adding:

```diff
def test_basic():

    ESM = """
    export function render(view) {
        view.el.innerText = "Hello, world";
    }
    """

    class Widget(AnyWidget):
        _esm = t.Unicode(ESM).tag(sync=True)

    w = Widget()

    assert w.has_trait("_esm"
++  assert w._esm == ESM

def test_legacy():

    ESM = """
    export function render(view) {
        view.el.innerText = "Hello, world";
    }
    """

    class Widget(AnyWidget):
        _module = t.Unicode(ESM).tag(sync=True)

    w = Widget()

    assert w.has_trait("_module")
++  assert w._module == ESM


++ def test_missing():
++     class Widget(AnyWidget):
++       ...
++
++    w = Widget()
++    assert w.has_trait("_esm")
++    assert w._esm == DEFAULT_ESM
```